src/test/java/com/xceptance/neodymium/tests/NeodymiumTestSelfTest.java

```diff
@@ -0,0 +1,177 @@
+package com.xceptance.neodymium.tests;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.Description;
+import org.junit.runner.Result;
+import org.junit.runner.notification.Failure;
+
+public class NeodymiumTestSelfTest extends NeodymiumTest
+{
+    @Rule
+    public TestName name = new TestName();
+
+    @Test
+    public void testCheckFailedOneFromOne() throws NoSuchFieldException, SecurityException, IllegalArgumentException, IllegalAccessException
+    {
+        final String errorMessage = "This is RuntimeException 1";
+        Result result = createResult(1, 0, new HashMap<String, Throwable>()
+        {
+            private static final long serialVersionUID = 1L;
+
+            {
+                this.put(name.getMethodName() + "1", new RuntimeException(errorMessage));
+            }
+        });
+        checkFail(result, 1, 0, 1, errorMessage);
+    }
+
+    @Test
+    public void testCheckFailedOneFromTwo() throws NoSuchFieldException, SecurityException, IllegalArgumentException, IllegalAccessException
+    {
+        final String errorMessage = "This is RuntimeException 1";
+        Result result = createResult(2, 0, new HashMap<String, Throwable>()
```

Comment:
See my comment above.

Reply:
Done

Reply:
Building the failures map can be simplified (similar to `testCheckFailedOneFromOne`)

Reply:
done